# Demographic analysis

In [1]:
import pandas as pd
import numpy as np

In [2]:
full_data_demo_duplicates = pd.read_csv('../merged_data/full_info_dataframe.csv', sep=",")

/var/folders/yr/gmmc3xcx5xg406qfnj2bnx340000gn/T/ipykernel_89853/1215530506.py:1: DtypeWarning: Columns (14) have mixed types. Specify dtype option on import or set low_memory=False.
  full_data_demo_duplicates = pd.read_csv('../merged_data/full_info_dataframe.csv', sep=",")


In [3]:
full_data_demo = full_data_demo_duplicates.sort_values('date_time').drop_duplicates(subset='client_id', keep='first')

In [4]:
full_data_demo['variation'].unique() 

array([nan, 'Control', 'Test'], dtype=object)

In [5]:
full_data_demo['variation'].value_counts()

variation
Test       26968
Control    23532
Name: count, dtype: int64

In [6]:
full_data_demo.groupby('variation')['clnt_age'].describe()

count       mean        std   min   25%   50%   75%   max
variation                                                             
Control    23526.0  47.498427  15.521781  17.0  33.5  48.5  60.0  96.0
Test       26961.0  47.163922  15.514461  17.0  33.5  47.5  59.5  96.0

In [7]:
full_data_demo['gendr'].unique()


array(['F', 'M', nan], dtype=object)

In [8]:
full_data_demo.groupby('variation')['gendr'].value_counts(normalize=True)

variation  gendr
Control    M        0.513763
           F        0.486237
Test       M        0.507376
           F        0.492624
Name: proportion, dtype: float64

## 🧪 Demographic Distribution Analysis: Test vs Control

---

### 📊 1. Sample Size

| Group   | Sample Size |
|---------|-------------|
| Test    | 26,968      |
| Control | 23,532      |

**Comment:**  
There is a slight difference in sample size (~3,400 more in the Test group), but it's not extreme. If statistical comparison is required, you might consider using stratified sampling or matching to balance the groups more precisely.

---

### 👤 2. Age Distribution (`clnt_age`)

| Group   | Mean  | Std. Dev. | Median | Min | Max |
|---------|-------|-----------|--------|-----|-----|
| Control | 47.50 | 15.52     | 48.5   | 17  | 96  |
| Test    | 47.16 | 15.51     | 47.5   | 17  | 96  |

**Comment:**  
Age distribution is highly consistent between the two groups. Means, medians, and ranges are nearly identical → ✅ **Balanced distribution.**

---

### 🚻 3. Gender Distribution (`gendr`)

| Group   | Male (%) | Female (%) |
|---------|----------|-------------|
| Control | 51.4%    | 48.6%       |
| Test    | 50.7%    | 49.3%       |

**Comment:**  
Gender distribution is very balanced across groups. There is no significant gender bias between Test and Control → ✅ **Balanced distribution.**

---

### ✅ Summary

The **Test** and **Control** groups are **demographically well balanced** in terms of:
- Sample size (reasonably close),
- Age (virtually identical distribution),
- Gender (no noticeable skew).

This indicates that the experimental design is solid from a demographic perspective.

---